src/backend/distributed/executor/local_executor.c

```diff
@@ -585,6 +586,46 @@ ExecuteLocalTaskPlan(PlannedStmt *taskPlan, char *queryString,
 }
 
 
+/*
+ * SetLocalExecutionStatus sets the local execution status to
+ * the given status, it errors if the transition is not possible from the
+ * current status.
+ */
+void
+SetLocalExecutionStatus(LocalExecutionStatus newStatus)
+{
+	if (!IsLocalExecutionStatusTransitionPossible(
+			CurrentLocalExecutionStatus,
+			newStatus))
+	{
+		ereport(ERROR,
+				(errmsg("cannot switch local execution status since it can cause "
+						"visibility problems in the current transaction")));
```

Comment:
Maybe inline the transition checks and give a relevant error message for each.

Reply:
yeah otherwise the error is subtle

Reply:
Bump